File: 1-two-sum/1-two-sum.py
class Solution:
    """
    1. We will search for nums[i]-target in map. If found we can directly return it that elements index and current index
    2. If not found then we can add that number to the map for future use
    """
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for i in range(len(nums)):
            if target-nums[i] in mp:
                return [i,mp[target-nums[i]]]
            mp[nums[i]] = i
        return -1

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
/*
    1. We will search for nums[i]-target in map. If found we can directly return it that elements index and current index
    2. If not found then we can add that number to the map for future use
*/
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            if (mp.find(target-nums[i])!=mp.end()){
                result.push_back(mp[target-nums[i]]);
                result.push_back(i);
                return result;
            }
            mp[nums[i]] = i;
        }
        return result;
        
    }
};